src/flaskapp/api.py

```diff
@@ -0,0 +1,226 @@
+        ):
+            return {"message": "Required info not found"}, 400
+        team_name = format_string(data["name"])
+        team_desc = format_string(data["desc"])
+        skills = []
+        if "skills" in data:
+            skills = format_string(data["skills"])
+        prizes = []
+        if "prizes" in data:
+            prizes = format_string(data["prizes"])
+        return create_team_profile(team_name, email, team_desc, skills, prizes)
+
+
+@app.route("/teams/<team_id>", methods=["GET", "PUT"])
+@ensure_user_logged_in
+def single_team(email, team_id):
+    if request.method == "GET":
+        return get_team_profile(email, team_id)
+
+    if request.method == "PUT":
+        return update_team_profile(email, team_id)
+
+
+@app.route("/teams/<team_id>/complete", methods=["PUT"])
+@ensure_user_logged_in
+def mark_team_complete(email, team_id):
+    return team_complete(email, team_id)
+
+
+@app.route("/teams/<team_id>/leave", methods=["PUT"])
+@ensure_user_logged_in
+def leave(email, team_id):
+    return user_leave(email, team_id)
+
+
+############################## UNIFY ##############################
+
+
+@app.route("/teams/<team_id>/invite", methods=["POST"])
+@ensure_user_logged_in
+def invite(email, team_id):
+    # NOTE team1 -inviting-> team2 (invite another team)
+    team1_name = team_id
+    data = request.get_json(silent=True)
+    if not data or "name" not in data or not data["name"]:
+        return {"message": "Required info not found"}, 400
+    team2_name = format_string(data["name"])
+    return team_invite(email, team1_name, team2_name)
+
+
+@app.route("/teams/<team_id>/confirm", methods=["POST"])
+@ensure_user_logged_in
+def confirm(email, team_id):
+    # NOTE team1 -confirms-> team2 (confirm an invite)
+    team1_name = team_id
+    data = request.get_json(silent=True)
+    if not data or "name" not in data or not data["name"]:
+        return {"message": "Required info not found"}, 400
+    team2_name = format_string(data["name"])
+    return team_confirm(email, team1_name, team2_name)
+
+
+@app.route("/teams/<team_id>/rescind", methods=["POST"])
+@ensure_user_logged_in
+def rescind(email, team_id):
+    # NOTE team1 -rescind-> team2 (rescind an invite)
+    team1_name = team_id
+    data = request.get_json(silent=True)
+    if not data or "name" not in data or not data["name"]:
+        return {"message": "Required info not found"}, 400
+    team2_name = format_string(data["name"])
+    return team_rescind(email, team1_name, team2_name)
+
+
+@app.route("/teams/<team_id>/reject", methods=["POST"])
+@ensure_user_logged_in
+def reject(email, team_id):
+    # NOTE team1 -reject-> team2 (rejecting an invite)
+    team1_name = team_id
+    data = request.get_json(silent=True)
+    if not data or "name" not in data or not data["name"]:
+        return {"message": "Required info not found"}, 400
+    team2_name = format_string(data["name"])
+    return team_reject(email, team1_name, team2_name)
+
```

Comment:
One thing to note while I was designing the Unify methods I overlooked the relationship where we may want to simulate a team that is interested to join another team. What we currently have is the relationship where a team is asking another team to join theirs. I think what we have here allows us to do what we want but it may feel weird for our end users. Food for thought.

Reply:
Tracking separately with https://github.com/HackRU/teamRU/issues/16